node/core/av-store/src/lib.rs

```diff
@@ -651,44 +671,49 @@ where
 	Context: SubsystemContext<Message=AvailabilityStoreMessage>
 {
 	use AvailabilityStoreMessage::*;
+
+	fn log_send_error(request: &'static str) {
+		log::debug!(target: LOG_TARGET, "error sending a response to {}", request);
+	}
+
 	match msg {
 		QueryAvailableData(hash, tx) => {
 			tx.send(available_data(&subsystem.inner, &hash).map(|d| d.data))
-				.map_err(|_| oneshot::Canceled)?;
+				.unwrap_or_else(|_| log_send_error("QueryAvailableData"));
 		}
 		QueryDataAvailability(hash, tx) => {
 			tx.send(available_data(&subsystem.inner, &hash).is_some())
-				.map_err(|_| oneshot::Canceled)?;
+				.unwrap_or_else(|_| log_send_error("QueryDataAvailability"));
 		}
 		QueryChunk(hash, id, tx) => {
 			tx.send(get_chunk(subsystem, &hash, id)?)
-				.map_err(|_| oneshot::Canceled)?;
+				.unwrap_or_else(|_| log_send_error("QueryChunk"));
 		}
 		QueryChunkAvailability(hash, id, tx) => {
 			tx.send(get_chunk(subsystem, &hash, id)?.is_some())
-				.map_err(|_| oneshot::Canceled)?;
+				.unwrap_or_else(|_| log_send_error("QueryChunkAvailability"));
 		}
 		StoreChunk { candidate_hash, relay_parent, validator_index, chunk, tx } => {
 			// Current block number is relay_parent block number + 1.
 			let block_number = get_block_number(ctx, relay_parent).await? + 1;
 			match store_chunk(subsystem, &candidate_hash, validator_index, chunk, block_number) {
 				Err(e) => {
-					tx.send(Err(())).map_err(|_| oneshot::Canceled)?;
+					tx.send(Err(())).unwrap_or_else(|_| log_send_error("StoreChunk"));
 					return Err(e);
 				}
 				Ok(()) => {
-					tx.send(Ok(())).map_err(|_| oneshot::Canceled)?;
+					tx.send(Ok(())).unwrap_or_else(|_| log_send_error("StoreChunk"));
 				}
 			}
 		}
 		StoreAvailableData(hash, id, n_validators, av_data, tx) => {
 			match store_available_data(subsystem, &hash, id, n_validators, av_data) {
 				Err(e) => {
-					tx.send(Err(())).map_err(|_| oneshot::Canceled)?;
+					tx.send(Err(())).unwrap_or_else(|_| log_send_error("StoreAvailableData"));
 					return Err(e);
 				}
 				Ok(()) => {
-					tx.send(Ok(())).map_err(|_| oneshot::Canceled)?;
+					tx.send(Ok(())).unwrap_or_else(|_| log_send_error("StoreAvailableData"));
```

Comment:
can we make these distinct? I guess, it may be useful to know in which branch this log was emitted exactly.

Reply:
I'd suggest either adding a `tag: &'static str` to the error variant or introducing more error variants.